src/snowflake/snowpark/modin/plugin/_internal/join_utils.py

```diff
@@ -1187,15 +1217,8 @@ def align(
     # NULL  NULL 2             NULL      4   e  2
     coalesce_key_config = None
     inherit_join_index = InheritJoinIndex.FROM_LEFT
-    # When it is `outer` align, we need to coalesce the align columns. However, if the
```

Comment:
@sfc-gh-nkumar this optimization is now performed in general for all join and align